swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll

```diff
@@ -46,7 +46,12 @@ module Generated {
     /**
      * Gets the `index`th spec of this availability info (0-based).
      */
-    final AvailabilitySpec getSpec(int index) { result = this.getImmediateSpec(index).resolve() }
+    final AvailabilitySpec getSpec(int index) {
+      exists(AvailabilitySpec immediate |
+        immediate = this.getImmediateSpec(index) and
+        if exists(this.getResolveStep()) then result = immediate else result = immediate.resolve()
+      )
```

Comment:
This additional and not-quite-trivial QL is repeated many, many times in the diff.  I suppose at code generation time we don't know whether `this.getResolveStep()` will exist or not, so we can't just generate one or the other case.  Still, I wonder if there's a way we can simplify / take out any common code here?

Reply:
I think one way to limit that is the other thing I propose doing, that is limit this kind of machinery to when the return type of the getter is an `Expr` or a `Pattern`. But not something to be tackled in this PR I think.

Reply:
:+1:

As long as the DCA run looks good performance-wise, I'm OK that we wait for that.